SimulationAgent/DeviceTelemetry/DeviceTelemetryActor.cs

```diff
@@ -151,10 +165,13 @@ public void HandleEvent(ActorEvents e)
                     this.ScheduleTelemetry();
                     break;
                 case ActorEvents.TelemetryDelivered:
```

Comment:
let's add a new event type called `ActorEvents.SendingTelemetry` and increment `this.totalMessageCount` only in there.  In `SendTelemetry.Run()` call `this.context.HandleEvent(DeviceTelemetryActor.ActorEvents.SendingTelemetry)` before sending the message